### DISCUSSION TITLE
get pagination quantity yourself

### DISCUSSION THREAD

**Discussion by Concentum** [laravel/framework#37769] [Ideas]
It would be nice if the ready-made total value could be passed to the paginate method of the builder.
Then you can very well customize requests to get the number of records without manually paginating

**Comment by kevin-rt** [comment]
I ran into a similar situation recently due to heavy database data I'm working with and trying to paginate.
My query is a bunch of left joins. The count query takes about 12 seconds, the offset query takes about half a second.
I'd get the same count by excluding my left joins, and get the count query down to half a second.
It would be great if I use a different query logic for the two queries to get speeds down to something more reasonable.

I saw this older discussion from 2017 and the mention of LengthAwarePaginator:
https://github.com/laravel/ideas/issues/826#issuecomment-335212357

```
$totalCount = /* logic goes here */;
$pageItems= $query->forPage($currentPage, $itemsPerPage)->get($columns);
$paginator = new LengthAwarePaginator($pageItems, $totalCount, $itemsPerPage, $currentPage)
```

A newer related discussion from a month ago where the author wants to pass in the page results as "data" then set the total calculations manually:
https://github.com/laravel/framework/discussions/41513
There's an idea there to extend LengthAwarePaginator with your own custom pagination logic, store the class in app/Pagination/CustomPaginator.php, then register it to your AppServiceProvider.

Probably could use either approach, or some combination of the two approaches to suit your needs.
